### PR TITLE
Add ability to draw "angle snapped" lines

### PR DESCRIPTION
**Pull request by JEphron**
Adds a new parameter to the line mode,`:brush/set line s <angle>`, which causes drawn lines to snap to increments of the given angle. 

https://user-images.githubusercontent.com/1817241/106367265-2ceae300-62f6-11eb-92bb-7de51cf46c71.mov

### CODE REVIEW

File: src/parser.rs [hunk 1]
```diff
@@ -188,6 +188,10 @@ impl Parse for LineDirection {
         choice(vec![
             string("v").value(Self::Vertical).label("vertical"),
             string("h").value(Self::Horizontal).label("horizontal"),
+            string("s")
```
Comment (JEphron) [inline src/parser.rs:191]: `s` for snap? `a` for angle? Something else entirely? 🤷

File: src/brush.rs [hunk 2]
```diff
@@ -199,6 +203,16 @@ impl Brush {
                 LineDirection::Free => self.curr,
                 LineDirection::Horizontal => Point2::new(self.curr.x, start.y),
                 LineDirection::Vertical => Point2::new(start.x, self.curr.y),
+                LineDirection::AngleSnap(snap) => {
```
Comment (JEphron) [inline src/brush.rs:206]: This does kind of make the Horizontal and Vertical modes redundant

Comment (cloudhead) [reply src/brush.rs:206]: Indeed. How about just having `:brush/set line` default to `0` (no snapping), and if you want horizontal or vertical you just use `90`? So then we get rid of `LineDirection` and just have an angle or `Option<u32>`?

Comment (JEphron) [reply src/brush.rs:200]: Makes sense. Removed.

File: src/brush.rs [hunk 3]
```diff
@@ -56,7 +41,10 @@ pub enum BrushMode {
     /// X-Ray mode.
     XRay,
     /// Confine stroke to a straight line from the starting point
-    Line(LineDirection),
+    Line(
+        /// snap angle (degrees)
+        Option<u32>,
+    ),
```
Comment (cloudhead) [inline src/brush.rs:47]:
```rust
Line { snap: Option<u32> }
```

**Comment by cloudhead** [comment]
Awesome. Works great!

For a nice mapping:
```
:map <shift> :brush/set line 45 {:brush/unset line}
```

**Pull request merged**